### ISSUE TITLE
Install STAR Executable Issue

### ISSUE THREAD

**Issue by Ayaldia** [alexdobin/STAR#166]
Hi, 

I am having trouble creating the STAR executable.
I am using Cygwin in Windows 10. 
This is the error message I am getting.

```
g++ -c   -O3 -pipe -std=c++11 -Wall -Wextra -fopenmp -D'COMPILATION_TIME_PLACE="Tue, Jun 28, 2016 11:12:50 AM LAPTOP-6CQM39D0:/cygdrive/e/Drop-Seq/STAR-2.5.2a/STAR-2.5.2a/source"'  SharedMemory.cpp
SharedMemory.cpp: In member function ‘void SharedMemory::CreateAndInitSharedObject(size_t)’:
SharedMemory.cpp:108:59: error: ‘SHM_NORESERVE’ was not declared in this scope
     _shmID=shmget(_key, toReserve, IPC_CREAT | IPC_EXCL | SHM_NORESERVE | 0666); //        _shmID = shmget(shmKey, shmSize, IPC_CREAT | SHM_NORESERVE | SHM_HUGETLB | 0666);
                                                           ^
SharedMemory.cpp: In member function ‘void SharedMemory::EnsureCounter()’:
SharedMemory.cpp:253:72: error: ‘SHM_NORESERVE’ was not declared in this scope
         _sharedCounterID=shmget(_counterKey, 1, IPC_CREAT | IPC_EXCL | SHM_NORESERVE | 0666);
                                                                        ^
make[1]: *** [Makefile:57: SharedMemory.o] Error 1
make[1]: Leaving directory '/cygdrive/e/Drop-Seq/STAR-2.5.2a/STAR-2.5.2a/source'
make: *** [Makefile:10: STAR] Error 2
```

Would greatly appreciate some advice!

Thanks!
Ayaldia

**Comment by Ayaldia** [comment]
Previous issue was avoided by removing #ifdef COMPILE_FOR_MAC and the #endif in the SharedMemory.cpp file

**Comment by alexdobin** [comment]
Hi Ayaldia,

were you able to run STAR under Cygwin?
Would be great if you can share your experience, I do not think this has been done before.

Cheers
Alex

**Comment by Ayaldia** [comment]
Hi,

Yes, as far as I can tell, STAR seems to be working in Cygwin. 
The main two issues I faced was `SHM_NORESERVE was not declared in this scope` 
and `vfork was not declared in this scope`
The SHM_NORESERVE issue was fixed by going into the SharedMemory.cpp file and deleting 
`#ifdef COMPILE_FOR_MAC` and `#endif`.
The vfork issue was fixed by going into the Parameters_openReadsFiles.cpp file and changing `vfork()` to `fork()`

I have only just started using STAR, but as far as I can tell it seems to be working fine.

Thanks!
Ayaldia
